RF-DETR does not automatically resize input images during training (resolution parameter seems ineffective)

### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.


### Bug

I am encountering an issue when training RF-DETR with images of varying resolutions. My dataset contains images with different sizes, such as 960×540 and 1920×1080. Models like YOLO and RT-DETR can automatically resize images (e.g., to 640×640) during training by setting an image_size or similar parameter. However, in RF-DETR, I observed that the input images are not automatically resized, which leads to the following error:

> AssertionError: Backbone requires input shape to be divisible by 32, but got torch.Size([8, 3, 1080, 1920])

I attempted the following approaches: 
```
model.train(resolution=640) 

model = RFDETRNano(resolution=640)
```
However, neither of these had any effect. The input images were still passed to the model in their original resolutions, and the same error persisted. 
Does RF-DETR support automatic image resizing during training (similar to YOLO or RT-DETR)?

### Environment

- RF-DETR: 0.1.0
- OS: Windows 10
- Python: 3.10.0
- PyTorch: 2.5.1
- GPU: NVIDIA RTX 4070ti super

### Minimal Reproducible Example

```
from rfdetr import RFDETRBase, RFDETRNano, RFDETRSmall, RFDETRMedium, RFDETRLarge

if __name__ == '__main__':
    model = RFDETRNano(resolution=640, num_classes=10)
    model.train(
        resolution=640
    )
```

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

@universeliang Hi, did you solve this issue?
I'm trying to finetune `medium` model with COCOA-cls, but I keep on getting the same error as you: `AssertionError: Backbone requires input shape to be divisible by 24, but got torch.Size([4, 3, 640, 640])`. I'm using 1.6.5post0 version.

--- a/tests/datasets/test_augmentations.py
+++ b/tests/datasets/test_augmentations.py
@@ -1507,6 +1507,80 @@ class TestMakeCocoTransformsAugConfig:
         assert len(wrappers) == expected_resize_wrappers
 
 
+class TestMakeCocoTransformsOutputSize:
+    """Regression tests for #979: transforms must resize high-resolution images to the target resolution.
+
+    These tests verify that ``make_coco_transforms`` and ``make_coco_transforms_square_div_64``
+    actually produce output images at the requested ``resolution``, not at the original image size.
+    Existing tests only check pipeline *structure*; these check actual output *dimensions*.
+    """
+
+    # 1920x1080 (landscape) — larger than any typical training resolution.
+    # PIL size is (width, height), so Image.new("RGB", (1920, 1080)) gives a 1920-wide, 1080-tall image.
+    _INPUT_W = 1920
+    _INPUT_H = 1080
+    _RESOLUTION = 640
+
+    def _make_image(self) -> Image.Image:
+        return Image.new("RGB", (self._INPUT_W, self._INPUT_H))
+
+    def test_square_val_resizes_large_image(self) -> None:
+        """Square val transform resizes 1920x1080 to exactly 640x640."""
+        transform = make_coco_transforms_square_div_64("val", self._RESOLUTION)
+        tensor, _ = transform(self._make_image(), None)
+        assert tensor.shape[-2:] == (self._RESOLUTION, self._RESOLUTION)
+
+    def test_square_train_resizes_large_image(self) -> None:
+        """Square train transform resizes 1920x1080 to 640x640 regardless of OneOf branch."""
+        transform = make_coco_transforms_square_div_64("train", self._RESOLUTION, aug_config={})
+        tensor, _ = transform(self._make_image(), None)
+        assert tensor.shape[-2:] == (self._RESOLUTION, self._RESOLUTION)
+
+    def test_nonsquare_val_longest_side_equals_1333(self) -> None:
+        """Non-square val transform on 1920x1080 produces longest side == 1333.
+
+        Pipeline: SmallestMaxSize(640) → 640x1138, then LongestMaxSize(1333) upscales → 750x1333.
+        LongestMaxSize resizes to exactly 1333 (not a cap), so the result always has max side == 1333
+        for inputs where the longest side after SmallestMaxSize is less than 1333.
+        """
+        transform = make_coco_transforms("val", self._RESOLUTION)
+        tensor, _ = transform(self._make_image(), None)
+        height, width = tensor.shape[-2], tensor.shape[-1]
+        assert max(height, width) == 1333
+
+    def test_nonsquare_val_longest_side_at_most_1333(self) -> None:
+        """Non-square val transform caps the longest side at 1333 px."""
+        transform = make_coco_transforms("val", self._RESOLUTION)
+        tensor, _ = transform(self._make_image(), None)
+        height, width = tensor.shape[-2], tensor.shape[-1]
+        assert max(height, width) <= 1333
+
+    def test_nonsquare_val_does_not_pass_original_dimensions(self) -> None:
+        """Non-square val transform must not emit the original 1920x1080 dimensions — the core regression."""
+        transform = make_coco_transforms("val", self._RESOLUTION)
+        tensor, _ = transform(self._make_image(), None)
+        height, width = tensor.shape[-2], tensor.shape[-1]
+        assert (height, width) != (self._INPUT_H, self._INPUT_W), (
+            f"Transform emitted original {self._INPUT_H}x{self._INPUT_W} — resize was not applied"
+        )
+
+    def test_square_val_does_not_pass_original_dimensions(self) -> None:
+        """Square val transform must not emit the original 1920x1080 dimensions — the core regression."""
+        transform = make_coco_transforms_square_div_64("val", self._RESOLUTION)
+        tensor, _ = transform(self._make_image(), None)
+        height, width = tensor.shape[-2], tensor.shape[-1]
+        assert (height, width) != (self._INPUT_H, self._INPUT_W), (
+            f"Transform emitted original {self._INPUT_H}x{self._INPUT_W} — resize was not applied"
+        )
+
+    def test_output_is_float_tensor(self) -> None:
+        """Transform pipeline produces a float32 tensor, not a PIL Image."""
+        transform = make_coco_transforms_square_div_64("val", self._RESOLUTION)
+        tensor, _ = transform(self._make_image(), None)
+        assert isinstance(tensor, torch.Tensor)
+        assert tensor.dtype == torch.float32
+
+
 class TestAugPresets:
     """Regression tests for built-in augmentation presets."""
 

--- a/tests/datasets/test_augmentations.py
+++ b/tests/datasets/test_augmentations.py
@@ -1552,8 +1552,8 @@ class TestMakeCocoTransformsOutputSize:
     def test_nonsquare_val_longest_side_at_most_1333(self) -> None:
         """Non-square val transform caps the longest side at 1333 px.
 
-        Use an input that still exceeds 1333 px on its longest side after SmallestMaxSize(640),
-        so this assertion specifically validates that LongestMaxSize(1333) is applied.
+        Use an input that still exceeds 1333 px on its longest side after SmallestMaxSize(640), so this assertion
+        specifically validates that LongestMaxSize(1333) is applied.
         """
         transform = make_coco_transforms("val", self._RESOLUTION)
         image = Image.new("RGB", (4000, 1000))

--- a/tests/datasets/test_augmentations.py
+++ b/tests/datasets/test_augmentations.py
@@ -1539,9 +1539,9 @@ class TestMakeCocoTransformsOutputSize:
     def test_nonsquare_val_resizes_and_caps_longest_side(self) -> None:
         """Non-square val transform resizes the image and keeps the longest side within 1333 px.
 
-        Avoid asserting an exact output dimension here because Albumentations resize behavior can vary
-        across supported versions. The stable contract is that the image is resized and the longest
-        side does not exceed the configured maximum.
+        Avoid asserting an exact output dimension here because Albumentations resize behavior can vary across supported
+        versions. The stable contract is that the image is resized and the longest side does not exceed the configured
+        maximum.
         """
         transform = make_coco_transforms("val", self._RESOLUTION)
         tensor, _ = transform(self._make_image(), None)

--- a/tests/datasets/test_augmentations.py
+++ b/tests/datasets/test_augmentations.py
@@ -1549,9 +1549,14 @@ class TestMakeCocoTransformsOutputSize:
         assert max(height, width) == 1333
 
     def test_nonsquare_val_longest_side_at_most_1333(self) -> None:
-        """Non-square val transform caps the longest side at 1333 px."""
+        """Non-square val transform caps the longest side at 1333 px.
+
+        Use an input that still exceeds 1333 px on its longest side after SmallestMaxSize(640),
+        so this assertion specifically validates that LongestMaxSize(1333) is applied.
+        """
         transform = make_coco_transforms("val", self._RESOLUTION)
-        tensor, _ = transform(self._make_image(), None)
+        image = Image.new("RGB", (4000, 1000))
+        tensor, _ = transform(image, None)
         height, width = tensor.shape[-2], tensor.shape[-1]
         assert max(height, width) <= 1333
 

--- a/tests/datasets/test_augmentations.py
+++ b/tests/datasets/test_augmentations.py
@@ -1536,17 +1536,18 @@ class TestMakeCocoTransformsOutputSize:
         tensor, _ = transform(self._make_image(), None)
         assert tensor.shape[-2:] == (self._RESOLUTION, self._RESOLUTION)
 
-    def test_nonsquare_val_longest_side_equals_1333(self) -> None:
-        """Non-square val transform on 1920x1080 produces longest side == 1333.
+    def test_nonsquare_val_resizes_and_caps_longest_side(self) -> None:
+        """Non-square val transform resizes the image and keeps the longest side within 1333 px.
 
-        Pipeline: SmallestMaxSize(640) → 640x1138, then LongestMaxSize(1333) upscales → 750x1333.
-        LongestMaxSize resizes to exactly 1333 (not a cap), so the result always has max side == 1333
-        for inputs where the longest side after SmallestMaxSize is less than 1333.
+        Avoid asserting an exact output dimension here because Albumentations resize behavior can vary
+        across supported versions. The stable contract is that the image is resized and the longest
+        side does not exceed the configured maximum.
         """
         transform = make_coco_transforms("val", self._RESOLUTION)
         tensor, _ = transform(self._make_image(), None)
         height, width = tensor.shape[-2], tensor.shape[-1]
-        assert max(height, width) == 1333
+        assert (height, width) != (self._INPUT_H, self._INPUT_W)
+        assert max(height, width) <= 1333
 
     def test_nonsquare_val_longest_side_at_most_1333(self) -> None:
         """Non-square val transform caps the longest side at 1333 px.

